python/cudf/cudf/_lib/parquet.pyx

```diff
@@ -11,15 +11,89 @@ from cudf._lib.utils cimport *
 from cudf._lib.utils import *
 from cudf._lib.includes.parquet cimport (
     reader as parquet_reader,
-    reader_options as parquet_reader_options
+    reader_options as parquet_reader_options,
+    compression_type,
+    write_parquet_args,
+    sink_info,
+    table_view,
+    table_metadata,
+    statistics_freq,
+    write_parquet as parquet_writer
 )
 from libc.stdlib cimport free
-from libcpp.memory cimport unique_ptr
+from libcpp.memory cimport unique_ptr, make_unique
 from libcpp.string cimport string
+from libcpp.map cimport map
 
 import errno
 import os
+import json
+import pandas as pd
+import pyarrow as pa
 
+from cudf._libxx.table cimport *
+
+
+cpdef generate_pandas_metadata(Table table, index):
```

Comment:
Make this a def and instead of strong typing as a Table we want to ensure it's a DataFrame otherwise it will fail weirdly in trying to get the indexes below.